src/test/java/task_3/test/CloudPricingCalculatorTest.java

```diff
@@ -1,71 +1,73 @@
 package task_3.test;
 
+import com.epam.training.olha_haichenkova.task_3.driver.DriverSingleton;
 import com.epam.training.olha_haichenkova.task_3.model.VirtualMachine;
-import com.epam.training.olha_haichenkova.task_3.page.GoogleCloudMainPage;
-import com.epam.training.olha_haichenkova.task_3.page.PricingCalculatorPage;
+import com.epam.training.olha_haichenkova.task_3.page.*;
+import com.epam.training.olha_haichenkova.task_3.page.fragment.pricing_calculator_page.CalculationResultsFragment;
+import com.epam.training.olha_haichenkova.task_3.service.TestDataReader;
 import com.epam.training.olha_haichenkova.task_3.service.VirtualMachineCreator;
 import com.epam.training.olha_haichenkova.task_3.util.TabsHandler;
-import com.epam.training.olha_haichenkova.task_3.page.YopmailMainPage;
+import com.epam.training.olha_haichenkova.task_3.util.CustomTestWatcher;
 
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.chrome.ChromeDriver;
 
+
+@ExtendWith(CustomTestWatcher.class)
 public class CloudPricingCalculatorTest {
 
-    public static WebDriver driver;
+    public WebDriver driver;
     private static final String SEARCH_QUERY = "Google Cloud Pricing Calculator";
     private static final int NUMBER_OF_VM = 4;
-    private static final String OPERATING_SYSTEM_SOFTWARE = "free";
-    private static final String VM_FAMILY = "gp";
-    private static final String VM_SERIES = "n1";
-    private static final String MACHINE_TYPE = "CP-COMPUTEENGINE-VMIMAGE-N1-STANDARD-8";
-    private static final String NUMBER_OF_GPU = "1";
-    private static final String GPU_TYPE = "NVIDIA_TESLA_V100";
-    private static final String LOCAL_SSD = "2";
-    private static final String DATACENTER_LOCATION = "europe-west3";
+    private static final String OPERATING_SYSTEM_SOFTWARE = "testdata.operatingSystemSoftware";
+    private static final String VM_FAMILY = "testdata.vMFamily";
+    private static final String VM_SERIES = "testdata.vmSeries";
+    private static final String MACHINE_TYPE = "testdata.machineType";
+    private static final String NUMBER_OF_GPU = "testdata.numberOfGPUs";
+    private static final String GPU_TYPE = "testdata.gPUType";
+    private static final String LOCAL_SSD = "testdata.localSSD";
+    private static final String DATACENTER_LOCATION = "testdata.datacenterLocation";
     private static final int COMMITTED_USAGE_YEARS = 1;
     private GoogleCloudMainPage googleCloudMainPage;
 
     @BeforeEach
     public void setUpDriver(){
-        driver = new ChromeDriver();
-        driver.manage().window().maximize();
+        driver = DriverSingleton.getDriver();
         googleCloudMainPage = new GoogleCloudMainPage(driver);
     }
```

Comment:
Could you move your Before/AfterEach methods to the Base test class so that your test classes extend it